### PR TITLE
chore: Remove unused code

### LINKED ISSUE #7080: Remove unused code
### Summary 
There are functions in the code base, which are not used anywhere, and can be removed. 
Sample when running
`GO111MODULE=on golangci-lint run --no-config --disable-all -E unused ./...`
```
...
pkg/github/github_app.go:28:22: func `(*GithubApp).isGithubAppEnabled` is unused (unused)
pkg/cmd/step/git/step_git.go:51:26: func `(*StepGitOptions).dropRepositories` is unused (unused)
pkg/cmd/get/get_build_logs.go:469:31: func `(*GetBuildLogsOptions).loadPipelineActivities` is unused (unused)
pkg/cmd/step/step_next_version.go:427:34: func `(*StepNextVersionOptions).setChartVersion` is unused (unused)
pkg/helm/helm_template.go:1057:24: func `(*HelmTemplate).getChartNameAndVersion` is unused (unused)
...
```

### PR DESCRIPTION
**Pull request by u5surf**
## Submitter checklist

- [x] Change is code complete and matches issue description.
- [x] Change is covered by existing or new tests.

#### Description
removing unused code to pass golangci-lint

#### Special notes for the reviewer(s)

golangci-lint result 
```
yugonoMacBook-Pro:jx yugo-horie$ GO111MODULE=on golangci-lint run --no-config --disable-all -E unused --timeout 600s ./...
yugonoMacBook-Pro:jx yugo-horie$
```

#### Which issue this PR fixes

fixes #7080

### CODE REVIEW

**Comment by hferentschik** [comment]
@u5surf, thanks for the pull requested. Much appreciated. I like the idea of removing unused code.

It seems all the status checks fail due to an unused import and the resulting compilation failure:

```
# github.com/jenkins-x/**/pkg/cmd/get
pkg/cmd/get/get_build_logs.go:9:2: imported and not used: "github.com/jenkins-x/**/pkg/apis/jenkins.io/v1"
make: *** [Makefile:90: build] Error 2
```

I think running `make ftm` should take care of this.

**Comment by ankitm123** [comment]
@u5surf, remember to enable `unused` linter, so that subsequent PRs wont re-introduce unused code :). https://github.com/jenkins-x/jx/blob/master/hack/linter.sh

File: pkg/github/github_app.go [hunk 1]
```diff
@@ -25,17 +25,6 @@ type response struct {
 	AppName      string
 }
 
-func (gh *GithubApp) isGithubAppEnabled() (bool, error) {
```
Comment (abayer) [inline pkg/github/github_app.go:28]: Revert this, please - this is used in a downstream dependency.

Comment (hferentschik) [reply pkg/github/github_app.go:28]: How can that be? A downstream dependency uses a non exported function?

**Comment by u5surf** [comment]
> @u5surf, thanks for the pull requested. Much appreciated. I like the idea of removing unused code.
> 
> It seems all the status checks fail due to an unused import and the resulting compilation failure:
> 
> ```
> # github.com/jenkins-x/**/pkg/cmd/get
> pkg/cmd/get/get_build_logs.go:9:2: imported and not used: "github.com/jenkins-x/**/pkg/apis/jenkins.io/v1"
> make: *** [Makefile:90: build] Error 2
> ```
> 
> I think running `make ftm` should take care of this.

File: pkg/github/github_app.go (hunk 1 above)
Comment (abayer) [reply pkg/github/github_app.go:28]: ah, yeah, I'm wrong, ignore me.

**Comment by u5surf** [comment]
> @u5surf, thanks for the pull requested. Much appreciated. I like the idea of removing unused code.
> 
> It seems all the status checks fail due to an unused import and the resulting compilation failure:
> 
> ```
> # github.com/jenkins-x/**/pkg/cmd/get
> pkg/cmd/get/get_build_logs.go:9:2: imported and not used: "github.com/jenkins-x/**/pkg/apis/jenkins.io/v1"
> make: *** [Makefile:90: build] Error 2
> ```
> 
> I think running `make ftm` should take care of this.

@hferentschik 
Thanks reviewing. Although I could compiled that one on my local environment, I have no idea why that CI failed?

```
$ make build
CGO_ENABLED=0 GO111MODULE=on go build -ldflags " -X github.com/jenkins-x/jx/pkg/version.Version=2.1.0-dev+cdfb07bb2 -X github.com/jenkins-x/jx/pkg/version.Revision='cdfb07bb2' -X github.com/jenkins-x/jx/pkg/version.Branch='issue-7080' -X github.com/jenkins-x/jx/pkg/version.BuildDate='20200422-22:02:27' -X github.com/jenkins-x/jx/pkg/version.GoVersion='1.13.1' """ -o build/jx cmd/jx/jx.go
$
```
I also cleaned modcache, but `make build` was passed.

c.f. go env
```
$ go env
GO111MODULE=""
GOARCH="amd64"
GOBIN=""
GOCACHE="/Users/yugo-horie/Library/Caches/go-build"
GOENV="/Users/yugo-horie/Library/Application Support/go/env"
GOEXE=""
GOFLAGS=""
GOHOSTARCH="amd64"
GOHOSTOS="darwin"
GONOPROXY=""
GONOSUMDB=""
GOOS="darwin"
GOPATH="/Users/yugo-horie/.go"
GOPRIVATE=""
GOPROXY="https://proxy.golang.org,direct"
GOROOT="/usr/local/go"
GOSUMDB="sum.golang.org"
GOTMPDIR=""
GOTOOLDIR="/usr/local/go/pkg/tool/darwin_amd64"
GCCGO="gccgo"
AR="ar"
CC="clang"
CXX="clang++"
CGO_ENABLED="1"
GOMOD="/Users/yugo-horie/.go/src/github.com/jenkins-x/jx/go.mod"
CGO_CFLAGS="-g -O2"
CGO_CPPFLAGS=""
CGO_CXXFLAGS="-g -O2"
CGO_FFLAGS="-g -O2"
CGO_LDFLAGS="-g -O2"
PKG_CONFIG="pkg-config"
GOGCCFLAGS="-fPIC -m64 -pthread -fno-caret-diagnostics -Qunused-arguments -fmessage-length=0 -fdebug-prefix-map=/var/folders/nc/h_bywb9j1118r0wlw5tdh6m80000gn/T/go-build316804223=/tmp/go-build -gno-record-gcc-switches -fno-common"

**Comment by hferentschik** [comment]
> Thanks reviewing. Although I could compiled that one on my local environment, I have no idea why that CI failed?

The code looks good. Do you mind trying a rebase of your code? See also https://github.com/jenkins-x/jx/pull/7117

**Comment by u5surf** [comment]
> > Thanks reviewing. Although I could compiled that one on my local environment, I have no idea why that CI failed?
> 
> The code looks good. Do you mind trying a rebase of your code? See also #7117

Sure, please.

**Comment by hferentschik** [comment]
Superseded by pull request #7117 which contains the same changes rebased on master.

**Pull request closed without merging**